src/main/java/hudson/plugins/gradle/injection/BuildScanEnvironmentContributor.java

```diff
@@ -23,15 +25,38 @@
 @Extension
 public class BuildScanEnvironmentContributor extends EnvironmentContributor {
 
+    private final ShortLivedTokenClient tokenClient;
+
+    public BuildScanEnvironmentContributor() {
+        this.tokenClient = new ShortLivedTokenClient();
+    }
+
+    public BuildScanEnvironmentContributor(ShortLivedTokenClient tokenClient) {
+        this.tokenClient = tokenClient;
+    }
+
     @Override
+    @SuppressFBWarnings("NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE")
     public void buildEnvironmentFor(@Nonnull Run run, @Nonnull EnvVars envs, @Nonnull TaskListener listener) {
         Secret secretKey = InjectionConfig.get().getAccessKey();
         Secret secretPassword = InjectionConfig.get().getGradlePluginRepositoryPassword();
         if ((secretKey == null && secretPassword == null) || alreadyExecuted(run)) {
             return;
         }
 
-        run.addAction(DevelocityParametersAction.of(new DevelocityLogger(listener), secretKey, secretPassword));
+        Secret shortLivedToken = null;
+        DevelocityLogger logger = new DevelocityLogger(listener);
+        if (secretKey != null && DevelocityAccessKey.isValid(secretKey.getPlainText())) {
+            shortLivedToken = tokenClient.get(InjectionConfig.get().getServer(),
+                    DevelocityAccessKey.parse(secretKey.getPlainText()),
```

Comment:
Unless I'm mistaken, this `parse` method assumes that the access key contains a single `server=key` entry, but the `isValid` method above will pass for a key with `server1=key1;server2=key2`

Reply:
You're right, then we need to extract the hostname from the DV server URL and try to find the corresponding key